packages/contracts-bridge/contracts/BridgeRouter.sol

```diff
@@ -244,16 +329,77 @@ contract BridgeRouter is Version0, Router {
         bytes29 _tokenId,
         bytes29 _action
     ) internal {
+        // tokens will be sent to the specified recipient
+        address _recipient = _action.evmRecipient();
+        // send tokens
+        _creditTokens(_origin, _nonce, _tokenId, _action, _recipient);
+        // dust the recipient with gas tokens
+        _dust(_recipient);
+    }
+
+    /**
+     * @notice Handles an incoming Connext Transfer message.
+     *
+     * Tokens are sent to the Connext Router,
+     * then Connext.reconcile callback is called
+     *
+     * @param _origin The domain of the chain from which the transfer originated
+     * @param _nonce The unique identifier for the message from origin to destination
+     * @param _tokenId The token ID
+     * @param _action The action
+     */
+    function _handleConnextTransfer(
+        uint32 _origin,
+        uint32 _nonce,
+        bytes29 _tokenId,
+        bytes29 _action
+    ) internal {
+        // tokens will be sent to connext router
+        address _recipient = address(connext);
+        // send tokens
+        address _token = _creditTokens(
+            _origin,
+            _nonce,
+            _tokenId,
+            _action,
+            _recipient
+        );
+        // call Connext reconcile callback
+        connext.reconcile(
+            _action.externalId(),
+            _action.amnt(),
+            _tokenId.id(),
+            _tokenId.domain(),
+            _token
+        );
+    }
+
+    /**
+     * @notice Credit tokens to a specified recipient.
+     * @dev Unlocks canonical tokens from escrow in BridgeRouter
+     * OR Mints representation tokens
+     * @param _origin The domain of the chain from which the transfer originated
+     * @param _nonce The unique identifier for the message from origin to destination
+     * @param _tokenId The canonical token identifier to credit
+     * @param _action The contents of the transfer message
+     * @param _recipient The recipient that will receive tokens
+     * @return _token The address of the local token contract
+     */
+    function _creditTokens(
+        uint32 _origin,
+        uint32 _nonce,
+        bytes29 _tokenId,
+        bytes29 _action,
+        address _recipient
+    ) internal returns (address _token) {
```

Comment:
this is a non-functional refactor, and should be in a separate PR before this one.

Reply:
(saying I approve of the changes, but they make the functional changes harder to review)

Reply:
https://github.com/nomad-xyz/monorepo/pull/385